### ISSUE TITLE
Programming: Python R and Female Programmers

### ISSUE THREAD

**Issue by sudocurse** [sudocurse/reading-list#7397]
Programming: Python, R, and Female Programmers<br>
http://ift.tt/2FphUcb<br>
<ul>
<li>
<h5><a href="https://opensource.com/article/18/3/getting-started-data-science" rel="nofollow">Getting started with Python for data science</a></h5>
<blockquote>
<p>Whether you're a budding data science enthusiast with a math or computer science background or an expert in an unrelated field, the possibilities data science offers are within your reach. And you don't need expensive, highly specialized enterprise software&mdash;the open source tools discussed in this article are all you need to get started.</p>
<p>Python, its machine-learning and data science libraries (pandas, Keras, TensorFlow, scikit-learn, SciPy, NumPy, etc.), and its extensive list of visualization libraries (Matplotlib, pyplot, Plotly, etc.) are excellent FOSS tools for beginners and experts alike. Easy to learn, popular enough to offer community support, and armed with the latest emerging techniques and algorithms developed for data science, these comprise one of the best toolsets you can acquire when starting out.</p>
</blockquote>
</li>
<li>
<h5><a href="https://opensource.com/article/18/3/r-programming-features" rel="nofollow">A glimpse into R counterculture</a></h5>
<blockquote>
<p>After many readers expressed their indignation, Milley wrote a follow-up blog post on the SAS website, which took on a considerably more diplomatic tone. She defended SAS as software that can be valued for its "support, reliability, and validation." Recent history, however, has made it much more difficult to conflate proprietary software with reliability or functionality.</p>
<p>R certainly presents a powerful case study in how an open source language has rendered long-dominant proprietary software, such as SAS, largely irrelevant. Although it is difficult to quantify the size of R's user base, one interesting metric of popularity is its use in academic journal articles. In that court, R surpassed SAS in 2015. Additionally, although it is merely anecdotal, it is amusing to note a thread from 2017 on the Statistics subreddit, in which the original poster wonders why SAS is still around in substantial numbers. To paraphrase the prevailing response, companies still buy SAS because it's what they have always used in the past and change is hard! Or as Woodrow Wilson put it, "If you want to make enemies, try to change something."</p>
<p>In contrast, there are developers and data science professionals who don't want to make any concessions to functionality. They want the optimal tools for their analyses, even if it means having to dig through Stack Overflow every now and then. For them, there is R. It started as a statistical computing environment, but it's had so many additions that it can now be classified as a general-purpose language.</p>
</blockquote>
</li>
<li>
<h5><a href="https://fossbytes.com/most-popular-programming-languages-female-programmers/" rel="nofollow">15 Most Popular Programming Languages Among Female Programmers</a></h5>
<blockquote>
<p>It&rsquo;s a known fact that there is a lack of gender diversity in the tech industry. While the companies and independent organizations are working to promote an open and inclusive environment, a lot of work needs to be done. However, a recent report from HackerRank suggests that things are slowly changing and the gender gap is slowly shrinking.</p>
<p>Named Women in Tech 2018, this report is based on the response from more than 14,000 professional developers. Out of them, about 2,000 were female. Before digging up and finding the most popular programming languages among female programmers, let me tell you some encouraging facts about the ongoing change.</p>
</blockquote>
</li>
</ul>
<p><a href="http://www.tuxmachines.org/node/109575" target="_blank">read more</a></p>
<br>
<br>
via Tux Machines http://ift.tt/VTl7oF<br>
March 05, 2018 at 03:55PM
